File: pythonperspective/question2.py
#The ever changing Ankara Designs
#You are  fashion desoigner known for your unique ans vibrant Ankara designs. Recently, you've discovered
#that some of your fabric patterns change their designs based on the temperature and mood of the wearer.
# You want to create a software that can predict the changes given the temperature and mood data. 
#Think about the classes you will need to model this changin Ankara and how to predict the changes.
class Migration:
    def __init__(self,animals,weather_patterns,river_levels,predator_location):
        self.animals = animals
        self.weather_patterns = weather_patterns
        self.river_levels = river_levels
        self.predator_location = predator_location
    def predictMovement(self):
        if self.weather_patterns=="wet" and self.river_levels > 5:
            return f"{self.animals} do not migrate"
        else:
            return f"{self.animals} can migrate"
    def predict_location (self,current_location):
        if current_location=="Mara" and self.predator_location == "Near":
            return f"{self.animals} migrate to Serengeti"
        elif current_location =="Serengeti" and self.predator_location =="Near":
            return f"{self.animals} migrate to Mara"
        else:
            return f"{self.animals} can stay in {current_location}"

File: question2.js
//The Great Migration
//Every year millions of wildebeest, zebras, and other animals participate in the Great Migration along
//trhe Serengeti Mara ecosysytem. As a coinservationist, you want to develop a software system that models
//this migraytion predicting the movement of the herds based on the weather patterns, riverlevels, and predator
//locations. Consider what classes you'll need to create to to represent the animals, environment, and
//various factors that influence the migration. 
class Migration{
    constructor(animals, weatherPatterns, riverLevels, predatorLocation){
    this.animals = animals
    this.weatherPatterns = weatherPatterns
    this.riverLevels = riverLevels
    this.predatorLocation = predatorLocation
    }
    predictMovement(){
        if (this.riverLevels >= 5 && this.weatherPatterns =="Wet"){
            console.log(this.animals + "will not migrate to Serengeti")
        }
        else if(this.riverLevels <5 && this.weatherPatterns =="Dry"){
            console.log(this.animals+"can cross")
        }
        else{
            console.log(this.animals+" can cross");
        }
    }
    location(currentLocation){
        if (currentLocation =="Mara"&& this.predatorLocation =="Near"){
            console.log(this.animals +" migrate to Serengeti")
        }
        else if (currentLocation=="Serengeti"&& this.predatorLocation == "Near"){
        console.log(this.animals + " Migrates to Mara")
        }
        else{
            console.log(this.animals + " can stay in "+{currentLocation})
        }
    }
}
one =new Migration(["Elephant", "Gazelle",],"Dry", 5,"Near")
one.predictMovement()
one.location("Mara")